benchmarks/circuit_construction.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Set
+
+import cirq
+
+
+def get_qubits(distance: int):
+    data_qubits: Set[cirq.Qid] = set()
+    z_measure_qubits: Set[cirq.Qid] = set()
+    x_measure_qubits: Set[cirq.Qid] = set()
+    for x in range(distance + 1):
+        for y in range(distance + 1):
+            if x < distance and y < distance:
+                data_qubits.add(cirq.GridQubit(2 * x + 1, 2 * y + 1))
+            on_boundary_x = (x == 0) or (x == distance)
+            on_boundary_y = (y == 0) or (y == distance)
+            parity = x % 2 != y % 2
+            if (on_boundary_x and parity) or (on_boundary_y and not parity):
+                continue
+            if parity:
+                z_measure_qubits.add(cirq.GridQubit(2 * x, 2 * y))
+            else:
+                x_measure_qubits.add(cirq.GridQubit(2 * x, 2 * y))
+    return data_qubits, z_measure_qubits, x_measure_qubits
+
+
+def rotated_surface_code_memory_z_cycle(
+    data_qubits, z_measure_qubits, x_measure_qubits, z_order, x_order
+) -> cirq.OP_TREE:
```

Comment:
Add docstring for this and other functions in this file.

Reply:
done